bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/LinkyHandlerFactory.java

```diff
@@ -50,69 +47,81 @@
+                                    LINKY_LOCALDATETIME_FORMATTER);
+                        } catch (Exception ex) {
+                            return LocalDate.parse(json.getAsJsonPrimitive().getAsString(), LINKY_LOCALDATE_FORMATTER)
+                                    .atStartOfDay();
+                        }
+                    })
             .create();
+
     private final LocaleProvider localeProvider;
-    private final HttpClient httpClient;
 
     @Activate
     public LinkyHandlerFactory(final @Reference LocaleProvider localeProvider,
-            final @Reference HttpClientFactory httpClientFactory) {
+            final @Reference HttpClientFactory httpClientFactory, final @Reference OAuthFactory oAuthFactory,
+            final @Reference HttpService httpService, final @Reference ThingRegistry thingRegistry,
+            ComponentContext componentContext) {
         this.localeProvider = localeProvider;
-        SslContextFactory sslContextFactory = new SslContextFactory.Client();
-        try {
-            SSLContext sslContext = SSLContext.getInstance("SSL");
-            sslContext.init(null, new TrustManager[] { TrustAllTrustManager.getInstance() }, null);
-            sslContextFactory.setSslContext(sslContext);
-        } catch (NoSuchAlgorithmException e) {
-            logger.warn("An exception occurred while requesting the SSL encryption algorithm : '{}'", e.getMessage(),
-                    e);
-        } catch (KeyManagementException e) {
-            logger.warn("An exception occurred while initialising the SSL context : '{}'", e.getMessage(), e);
-        }
-        this.httpClient = httpClientFactory.createHttpClient(LinkyBindingConstants.BINDING_ID, sslContextFactory);
-        httpClient.setFollowRedirects(false);
-        httpClient.setRequestBufferSize(REQUEST_BUFFER_SIZE);
-    }
-
-    @Override
-    protected void activate(ComponentContext componentContext) {
-        super.activate(componentContext);
-        try {
-            httpClient.start();
-        } catch (Exception e) {
-            logger.warn("Unable to start Jetty HttpClient {}", e.getMessage());
-        }
-    }
 
-    @Override
-    protected void deactivate(ComponentContext componentContext) {
-        super.deactivate(componentContext);
-        try {
-            httpClient.stop();
-        } catch (Exception e) {
-            logger.warn("Unable to stop Jetty HttpClient {}", e.getMessage());
-        }
+        this.httpClientFactory = httpClientFactory;
+        this.oAuthFactory = oAuthFactory;
+        this.httpService = httpService;
+        this.thingRegistry = thingRegistry;
+        this.componentContext = componentContext;
     }
 
     @Override
     public boolean supportsThingType(ThingTypeUID thingTypeUID) {
-        return THING_TYPE_LINKY.equals(thingTypeUID);
+        return LinkyBindingConstants.SUPPORTED_DEVICE_THING_TYPES_UIDS.contains(thingTypeUID);
```

Comment:
maybe you could `import static org.openhab.binding.linky.internal.LinkyBindingConstants.*`

Reply:
fix in next commit